packages/design-system/src/components/tooltip/index.js

```diff
@@ -106,6 +106,7 @@ function Tooltip({
   const [arrowDelta, setArrowDelta] = useState(null);
   const anchorRef = useRef(null);
   const tooltipRef = useRef(null);
+  const dynamicPlacement = useRef(null);
```

Comment:
What's the rationale for using a ref here instead of state? I guess I only worry that this can update and not cause a rerender causing `TooltipContainer` to take a stale `placement` prop

Reply:
I didn't want `Tooltip` to _have_ to rerender when the `dynamicPlacement` isn't necessary, and since `Popup` and `ToolTipContainer` will rerender when the `placement` prop changes anyway, it would always getting the updated / correct value for `placement`. At least I think.

Reply:
hmm you are correct, I am getting a stale state for placement. I update to use  `useState` 🙌🏽